src/main/java/cuchaz/enigma/gui/elements/CovertTextField.java

```diff
@@ -0,0 +1,170 @@
+package cuchaz.enigma.gui.elements;
+
+import java.awt.GridLayout;
+import java.awt.event.*;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.swing.BorderFactory;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.text.Document;
+
+import cuchaz.enigma.gui.events.CovertTextFieldListener;
+import cuchaz.enigma.utils.Utils;
+import cuchaz.enigma.utils.validation.ParameterizedMessage;
+import cuchaz.enigma.utils.validation.Validatable;
+
+/**
+ * A label that converts into an editable text field when you click it.
+ */
+public class CovertTextField implements Validatable {
```

Comment:
There's a typo here. Also, I think `Converting` would be better than `Convert`.

Reply:
No, that's not a typo, it's just a horrible name for what this has become (well, it has always been a horrible name). `Converting` is good actually, I think I'll go with that then.